tardis/io/logger/logger.py

```diff
@@ -1,165 +1,197 @@
-    """
-    if "debug" in tardis_config:
-        specific_log_level = (
-            tardis_config["debug"]["specific_log_level"]
-            if specific_log_level is None
-            else specific_log_level
-        )
-
-        logging_level = (
-            log_level if log_level else tardis_config["debug"]["log_level"]
-        )
-
-        # Displays a message when both log_level & tardis["debug"]["log_level"] are specified
-        if log_level and tardis_config["debug"]["log_level"]:
-            print(
-                "log_level is defined both in Functional Argument & YAML Configuration {debug section}"
+import re
+from ipywidgets import Output, Tab, Layout
+from IPython.display import display, HTML
+from dataclasses import dataclass, field
+
+@dataclass
+class LoggingConfig:
+    LEVELS: dict[str, int] = field(default_factory=lambda: {
+        "NOTSET": logging.NOTSET,
+        "DEBUG": logging.DEBUG,
+        "INFO": logging.INFO,
+        "WARNING": logging.WARNING,
+        "ERROR": logging.ERROR,
+        "CRITICAL": logging.CRITICAL,
+    })
+
+    COLORS: dict[int | str, str] = field(default_factory=lambda: {
+        logging.INFO: "#D3D3D3",
+        logging.WARNING: "orange",
+        logging.ERROR: "red",
+        logging.CRITICAL: "orange",
+        logging.DEBUG: "blue",
+        "default": "black",
+    })
+    
+    DEFAULT_LEVEL = "INFO"
+    DEFAULT_SPECIFIC_STATE = False
+
+
+LOGGING_LEVELS = LoggingConfig().LEVELS
+
+class TardisLogger:
+    def __init__(self):
+        self.config = LoggingConfig()
+        self.log_outputs = {
+            "WARNING/ERROR": self._create_output_widget(),
+            "INFO": self._create_output_widget(),
+            "DEBUG": self._create_output_widget(),
+            "ALL": self._create_output_widget(),
+        }
+        self.logger = logging.getLogger("tardis")
+        
+    def _create_output_widget(self, height="300px"):
+        return Output(layout=Layout(height=height, overflow_y="auto"))
+    
+    def configure_logging(self, log_level, tardis_config, specific_log_level=None):
+        if "debug" in tardis_config:
+            specific_log_level = tardis_config["debug"].get(
+                "specific_log_level", specific_log_level
             )
-            print(
-                f"log_level = {log_level.upper()} will be used for Log Level Determination\n"
+            logging_level = log_level or tardis_config["debug"].get(
+                "log_level", "INFO"
             )
-
-    else:
-        # Adds empty `debug` section for the YAML
-        tardis_config["debug"] = {}
-
-        if log_level:
-            logging_level = log_level
+            if log_level and tardis_config["debug"].get("log_level"):
+                print(
+                    "log_level is defined both in Functional Argument & YAML Configuration {debug section}"
+                )
+                print(
```

Comment:
Slightly odd for this to be two print statements. Also, even though this refers to the logger, I would still think this would be a logging statement and not a print message. Maybe just at Critical or Error level so that it always displays.